monitor.cpp

```diff
@@ -5,34 +5,40 @@
 #include <iostream>
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
+bool isTemperatureCritical(float temperature) {
+  return (temperature > 102 || temperature < 95);
+}
+
+bool isPulseRateCritical(float pulseRate) {
+  return (pulseRate < 60 || pulseRate > 100);
+}
+
+bool isSpo2Critical(float spo2) {
+  return (spo2 < 90);
+}
+
+void displayWarning(const std::string& message) {
+  cout << message << "\n";
+  for (int i = 0; i < 6; i++) {
+    cout << "\r* " << flush;
+    sleep_for(seconds(1));
+    cout << "\r *" << flush;
+    sleep_for(seconds(1));
   }
-  return 1;
+}
+
+int handleAndCheckVitals(const std::string& message, bool isCritical) {
+    if (isCritical) {
+        displayWarning(message);
+        return 0;
+    }
+    return 1;
+}
+
+int vitalsOk(float temperature, float pulseRate, float spo2) {
+    int result = 1;
+    result &= handleAndCheckVitals("Temperature is critical!", isTemperatureCritical(temperature));
```

Comment:
This is a binary &, not a logical &. Perhaps that's why the function returns int and not bool.
Try `std::any_of`. Does it feel more natural?